test/integration/integration.cc

```diff
@@ -813,6 +813,7 @@ void BaseIntegrationTest::testUpstreamProtocolError() {
   IntegrationCodecClientPtr codec_client;
   IntegrationStreamDecoderPtr response(new IntegrationStreamDecoder(*dispatcher_));
   FakeRawConnectionPtr fake_upstream_connection;
+  fake_upstreams_[0]->setWaitForHttpConnection(false);
```

Comment:
I'm a little confused by this part of the change. It seems fragile that we have to remember to set this outside the test infra. Why is this needed? And we still call makeHttpConnection() below?